managedcache/trackingcache.go

```diff
@@ -455,3 +476,60 @@ func (c *trackingCache) RemoveOtherInformers(ctx context.Context, gvks sets.Set[
 		return ctx.Err()
 	}
 }
+
+func (c *trackingCache) Watch(ctx context.Context, user client.Object, gvks sets.Set[schema.GroupVersionKind]) error {
+	if err := c.watch(ctx, gvks); err != nil {
+		return err
+	}
+
+	c.watchesByUserLock.Lock()
+	defer c.watchesByUserLock.Unlock()
+	c.watchesByUser[toAccessManagerKey(user)] = gvks
```

Comment:
Is there a potential race condition here where the same user requests two different sets of GVKs, the both call `c.watch` successfully, but then only one one set of GVKs is added to `c.watchesByUser`?